### PR TITLE
Use Cloud Datastore to store the tweet ID and status

### PR DESCRIPTION
**Pull request by dlsniper**

### CODE REVIEW

File: bot/bot.go [hunk 1]
```diff
@@ -12,7 +12,9 @@ import (
 	"strconv"
 	"strings"
 
+	"cloud.google.com/go/datastore"
 	"github.com/nlopes/slack"
+	"golang.org/x/net/context"
```
Comment (judwhite) [inline bot/bot.go]: Can you import `"context"`, or do you need to support pre-1.7?

File: bot/gerrit.go [hunk 2]
```diff
@@ -26,41 +26,76 @@ type (
...
+		}
+		if key == nil {
+			return -1, nil
+		}
+		return int(key.ID()), nil
+	}()
+
+	if err != nil {
+		b.logf("got error while loading last ID from the datastore: %v\n", err)
 	}
```
Comment (judwhite) [inline bot/gerrit.go]: keep going, or return?

File: bot/gerrit.go [hunk 3]
```diff
@@ -26,41 +26,76 @@ type (
...
-		for _, msg := range history.Messages {
-			if msg.User != b.id {
-				continue
-			}
-			if len(msg.Attachments) != 1 {
-				continue
-			}
-			lastID = strings.ToLower(msg.Attachments[0].Footer)
-			break
+		dst := &goCL{}
```
Comment (judwhite) [inline bot/gerrit.go]: nit: could `var dst goCL` and `&dst` as necessary

File: bot/gerrit.go [hunk 4]
```diff
@@ -26,41 +26,76 @@ type (
 			} `json:"commit"`
 		} `json:"revisions"`
 	}
+
+	goCL struct {
+		Tweeted   bool `datastore:",noindex"`
+		CrawledAt time.Time
+	}
```
Comment (judwhite) [inline bot/gerrit.go]: nit: I understand it's not required, but may want to consider giving an explicit `datastore:"name"` to be explicit and prevent issues if the struct field names are ever refactored.

File: bot/gerrit.go (hunk 3 above)
Comment (dlsniper) [inline bot/gerrit.go]: it seems simpler than to do it like this: 
```go
	getCLFromDS := func(query *datastore.Query) (*datastore.Key, *goCL, error) {
		iter := b.dsClient.Run(b.ctx, query)

		var dst goCL
		key, err := iter.Next(&dst)
		if err != nil && err != datastore.Done {
			b.logf("error while fetching history: %v\n", err)
			return nil, nil, err
		}

		return key, &dst, nil
	}
```

File: bot/gerrit.go [hunk 5]
```diff
@@ -87,7 +122,7 @@ func (b *Bot) MonitorGerrit(duration time.Duration) {
```
Comment (dlsniper) [inline bot/gerrit.go]: Done, thanks.

File: bot/gerrit.go (hunk 4 above)
Comment (dlsniper) [inline bot/gerrit.go]: Done, thanks.

File: bot/gerrit.go (hunk 2 above)
Comment (dlsniper) [inline bot/gerrit.go]: ` return `, def ` return `, thanks.

**Pull request merged**